packages/tradershub/src/components/Dialog/Dialog.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { ReactElement } from 'react';
+import { useEventListener } from 'usehooks-ts';
+import { qtMerge } from '@deriv/quill-design';
+import { useModal } from '../ModalProvider';
+import DialogAction from './DialogAction';
+import DialogContent from './DialogContent';
+import DialogHeader from './DialogHeader';
+
+/**
+ * Type for the Dialog children
+ * @typedef TDialogChildren
+ */
+type TDialogChildren =
+    | ReactElement<typeof DialogAction>
+    | ReactElement<typeof DialogContent>
+    | ReactElement<typeof DialogHeader>;
+
+/**
+ * Type for the Dialog component props
+ * @typedef TDialog
+ * @property {TDialogChildren | TDialogChildren[]} children - Children nodes
+ * @property {string} [className] - Optional CSS class name
+ * @property {boolean} [shouldPreventCloseOnEscape] - Optional flag to prevent closing the dialog on Escape key press
+ */
+type TDialog = {
+    children: TDialogChildren | TDialogChildren[];
+    className?: string;
+    shouldPreventCloseOnEscape?: boolean;
```

Comment:
will it have a clickOnBackdrop too?

Reply:
According @thisyahlen-deriv there's seems to be an issues with the `clickOnBackdrop` because we are using values `@deriv/quill-design`. We will address the issue in another PR 😃 👍🏼